### PR TITLE
operation logs: migrate legacy operation logs

### PR DESCRIPTION
**Pull request by BadrAly**
This cli is used only in the reroils instance to migrate
Virtua create and update operation logs from the rero
production database.

Co-Authored-by: Aly Badr <aly.badr@rero.ch>

## Why are you opening this PR?

https://tree.taiga.io/project/rero21-reroils/task/2007?kanban-status=1224893

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

File: rero_ils/modules/operation_logs/cli.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+from rero_ils.modules.operation_logs.models import OperationLogOperation
+from rero_ils.modules.utils import extracted_data_from_ref
+
+
+@click.command('migrate_virtua_operation_logs')
+@click.option('-f', 'input_file', help='Logs input file')
+@with_appcontext
+def migrate_virtua_operation_logs(input_file):
+    """Migrate Virtua operation logs to REROILS."""
+    with open(input_file, 'r', encoding='utf-8') as request_file:
```
Comment (rerowep) [inline rero_ils/modules/operation_logs/cli.py:37]: If the file is always needed it is better to make it a parameter:
`@click.argument('input_file', type=click.File('r'))`Click can open the file already.
If the file can be big, it is better to implement also lazy loading for the json file.
Verbose should be nice to see an feedback if needed.

File: rero_ils/modules/operation_logs/cli.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                    oplg.get('record').get('$ref'), data='pid')
+                resource = extracted_data_from_ref(
+                    oplg.get('record').get('$ref'), data='resource')
+                create_rec = \
+                    OperationLog.get_create_operation_log_by_resource_pid(
+                        resource, record_pid)
+                if create_rec:
+                    create_rec.update(oplg, dbcommit=True, reindex=True)
+            elif operation == OperationLogOperation.UPDATE:
+                    OperationLog.create(data=oplg, dbcommit=True, reindex=True)
```
Comment (rerowep) [inline rero_ils/modules/operation_logs/cli.py:52]: We have `OperationLog.create` but we in `operationLogOperation.UPDATE` ?

File: rero_ils/modules/operation_logs/cli.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            if operation == OperationLogOperation.CREATE:
+                record_pid = extracted_data_from_ref(
+                    oplg.get('record').get('$ref'), data='pid')
+                resource = extracted_data_from_ref(
+                    oplg.get('record').get('$ref'), data='resource')
+                create_rec = \
+                    OperationLog.get_create_operation_log_by_resource_pid(
+                        resource, record_pid)
+                if create_rec:
+                    create_rec.update(oplg, dbcommit=True, reindex=True)
```
Comment (rerowep) [inline rero_ils/modules/operation_logs/cli.py:50]: We have `OperationLog. update ` but we in `operationLogOperation. CREATE ` ?

File: scripts/setup [hunk 4]
```diff
@@ -483,6 +483,14 @@ then
     diff --suppress-common-lines --side-by-side tmp/mapping_before_setup.txt tmp/mapping_after_setup.txt
 fi
 
+info_msg "- Load Virtua CREATE operation logs: ${DATA_PATH}/operation_logs_creations.json"
+eval ${PREFIX} invenio fixtures migrate_virtua_operation_logs -f ${DATA_PATH}/operation_logs_creations.json
+info_msg "- Load Virtua UPDATE operation logs: ${DATA_PATH}/operation_logs_updates.json"
+eval ${PREFIX} invenio fixtures migrate_virtua_operation_logs -f ${DATA_PATH}/operation_logs_updates.json
+
```
Comment (rerowep) [inline scripts/setup:490]: Why are the update and creation operation logs in separated files?

**Review by rerowep — CHANGES_REQUESTED** [review]

File: rero_ils/modules/operation_logs/cli.py (hunk 3 above)
Comment (BadrAly) [reply rero_ils/modules/operation_logs/cli.py]: It is normal, I added a code comment to explain this

File: scripts/setup (hunk 4 above)
Comment (BadrAly) [reply scripts/setup:488]: In plexus, you have now one file to load `operation_logs.json`

**Review by iGormilhit — CHANGES_REQUESTED** [review]
Commit message proposition:

```
operation logs: migrate legacy operation logs

* Adds a CLI, used only in the RERO ILS instance, to migrate Virtua
  'create' and 'update' operation logs from the RERO legacy production
  database (Virtua).
```

**Review by iGormilhit — APPROVED** [review]

File: rero_ils/modules/operation_logs/cli.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+                    pid_type, record_pid)
+            if create_rec:
+                create_rec.update(oplg, dbcommit=True, reindex=True)
+                count += 1
+        elif pid_type and operation == OperationLogOperation.UPDATE:
+            # The virtua update operation log is a new entry in the reroils
+            # operation log, the method to use is CREATE
+            OperationLog.create(data=oplg, dbcommit=True, reindex=True)
+            count += 1
+    click.echo(
```
Comment (rerowep) [inline rero_ils/modules/operation_logs/cli.py:78]: Could you add verbose printing of operation: `count type operation id` as we do for the other resources?

**Review by rerowep — CHANGES_REQUESTED** [review]

File: rero_ils/modules/operation_logs/cli.py (hunk 5 above)
Comment (BadrAly) [reply rero_ils/modules/operation_logs/cli.py]: -> progress bar

**Review by lauren-d — APPROVED** [review]

**Review by rerowep — APPROVED** [review]

**Pull request merged**